### PR TITLE
nix: reafactoring of status-go builds

### PR DESCRIPTION
**Pull request by jakubgs**
This is a refactor of `status-go` builds done with Nix.
Now they can be made using simply one of the new `Makefile` targets starting with `status-go-*`:
```
 > make status-go-desktop

nix/scripts/build.sh targets.status-go.desktop
Running: nix-build --pure --fallback --no-out-link --show-trace --attr targets.status-go.desktop  default.nix

Extracting result: /nix/store/p22sdsvrqb54x4mpbvqws0658r1szwm7-status-go-0.52.4-7bbe956
'/nix/store/p22sdsvrqb54x4mpbvqws0658r1szwm7-status-go-0.52.4-7bbe956/include/libstatus.h' -> './result/include/libstatus.h'
'/nix/store/p22sdsvrqb54x4mpbvqws0658r1szwm7-status-go-0.52.4-7bbe956/lib/x86_64-linux/libstatus.a' -> './result/lib/x86_64-linux/libstatus.a'

SUCCESS
```
And I also fixed the shells for desktop development:
```
 > make shell TARGET=desktop

[nix-shell:~/work/status-react]$ echo $STATUS_GO_DESKTOP_LIBDIR
/nix/store/p22sdsvrqb54x4mpbvqws0658r1szwm7-status-go-0.52.4-7bbe956/lib

[nix-shell:~/work/status-react]$ echo $QT_PATH 
/nix/store/jsbg2bp3cwrga898l5lki0sm13gz0rxb-qt-full-5.12.7

[nix-shell:~/work/status-react]$ echo $QTKEYCHAIN_SOURCES
/nix/store/xrh0pywpa0kcm6rvy5yspzji7pvgvqah-qtkeychain-patched-source/src
```

With this merged I'll be able to improve the desktop PR with @siphiuel: #10411

Changes:
- Fixed `nix/status-go/desktop` builds
- Dropped nimbus wrapper for `status-go` for now
- Split `status-go` builds into subfolders: `mobile`, `desktop`
- Fixed shells for desktop builds: `linux`,`macos`,`windows`
- Added `make status-go-*` targets for building them
- Moved source management to `nix/status-go/source.nix`
- Moved `nix/status-go/build.nix` into `nix/status-go/mobile`
- Dropped `nix/tools/mkShell.nix` since it did almost nothing
- Moved setup from `nix/tools/mkShell.nix` to `nix/shells.nix`
- Moved `nix/status-go/build.nix` into `nix/status-go/mobile`
- Moved `nix/desktop/cmake/qtkeychain` to `nix/pkgs/qtkeychain-src`
- Moved `nix/desktop/linux/linuxdeployqt` to `nix/pkgs`
- Moved `nix/desktop/linux/appimagekit` to `nix/pkgs`
- Simplified `nix/mobile/ios/status-go-shell.nix`
- Simplified `nix/status-go/default.nix`
- Updated the `nix/DETAILS.md` and `nix/README.md`
- Moved known issues to `nix/KNOWN_ISSUES.md`
- Improved output of `nix/scripts/build.sh`

Status: __READY__

### CODE REVIEW

**Comment by jakubgs** [comment]
I need to still re-add the Nimbus stuff which I avoided on purpose to make the refactor simpler.

**Comment by jakubgs** [comment]
I just found out while trying to re-add the Nimbus wrapper stuff for `status-go` that Pedro's branch in `nimbus` repo was never merged into `master`:
https://github.com/status-im/nimbus/tree/feature/android-api
Last commit is from 4th of January and it doesn't seem to work. At least the submodules seem fucked:
```
From https://github.com/status-im/secp256k1
 * [new branch]      master     -> origin/master
fatal: reference is not a tree: be6f5385330905bf1d7cc441be6703cfa7aef847
Unable to checkout be6f5385330905bf1d7cc441be6703cfa7aef847 from https://github.com/status-im/secp256k1.git.
```
Not sure what to do about that. If we want this to work it might make more sense to do a separate PR that tries to fix this properly. For now I can leave out the Nimbus stuff.

File: nix/desktop/default.nix [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
   };
+
+  # for merging default shell
+  mergeDefaultShell = (key: val: { shell = mergeSh default [ val.shell ]; });
+
+in {
+  shell = default;
 }
+  # merge default shell with platform sub-shells
+  // mapAttrs mergeDefaultShell { inherit linux windows macos; }
```
Comment (yenda) [inline nix/desktop/default.nix:49]: // is a comment or some syntax?

**Review by yenda — APPROVED** [review]

File: nix/desktop/default.nix (hunk 1 above)
Comment (jakubgs) [reply nix/desktop/default.nix:49]: The `//` syntax is for merging attribute sets:
```
nix-repl> { a = 1; } // { b = 2; }
{ a = 1; b = 2; }
```

Comment (jakubgs) [reply nix/desktop/default.nix:49]: https://nixos.wiki/wiki/Nix_Expression_Language#Operators

**Review by vitvly — APPROVED** [review]
+++

**Pull request merged**